src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -100,35 +418,67 @@ public View(ColonyView c, BlockPos l)
         @Override
         public void deserialize(ByteBuf buf)
         {
+            fields = new ArrayList<>();
             super.deserialize(buf);
+            assignFieldManually = buf.readBoolean();
+            int size = buf.readInt();
+            for(int i = 1; i <= size; i++)
+            {
+                FieldView fieldView = new FieldView();
+                fieldView.deserialize(buf);
+                fields.add(fieldView);
+                if(fieldView.isTaken())
+                {
+                    amountOfFields++;
+                }
+            }
+            workerName = ByteBufUtils.readUTF8String(buf);
+        }
 
-            wheat = buf.readInt();
-            potato = buf.readInt();
-            carrot = buf.readInt();
-            melon = buf.readInt();
-            pumpkin = buf.readInt();
+        /**
+         * Should the farmer be assigned manually to the fields?
+         * @return true if yes.
+         */
+        public boolean assignFieldManually()
+        {
+            return assignFieldManually;
         }
-    }
 
-    @Override
-    public void serializeToView(ByteBuf buf)
-    {
-        super.serializeToView(buf);
+        /**
+         * Getter of the fields list.
+         * @return an unmodifiable List.
+         */
+        public List<FieldView> getFields()
+        {
+            return Collections.unmodifiableList(fields);
+        }
 
-        buf.writeInt(wheat);
-        buf.writeInt(potato);
-        buf.writeInt(carrot);
-        buf.writeInt(melon);
-        buf.writeInt(pumpkin);
-    }
+        /**
+         * Getter of the worker name.
+         * @return the name of the worker.
+         */
+        public String getWorkerName()
+        {
+            return workerName;
+        }
 
-    /**
-     * Returns the farm radius of the building
-     *
-     * @return      Farm radius
-     */
-    public int getFarmRadius()
-    {
-        return getBuildingLevel()+3;
+        /**
+         * Getter for amount of fields.
+         * @return the amount of fields.
+         */
+        public int getAmountOfFields()
+        {
+            return amountOfFields;
+        }
+
+        /**
+         * Reduces the amount of fields by i.
+         * @param i the number to reduce.
+         */
+        public void reduceAmountOfFields(final int i)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "i". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)